### ISSUE TITLE
The Apache Camel not being able to find the Kamelet with the ID jms-apache-artemis-source

### ISSUE THREAD

**Issue by RafiIsaharov** [apache/camel-karavan#1042]
The camel-karavan service was installed in minikube v1.32.0 with the docker driver on Ubunto 22.04 via WSL2 on Windows 10
created a simple project from https://github.com/apache/camel-karavan/tree/main/karavan-demo/jms-to-kafka

When running via the Karavan UI dev mode the example from the link above, getting an error:

```
2024-01-01 09:43:39.856  WARN 15 --- [ ReloadOnDemand] support.RouteOnDemandReloadStrategy : Error reloading routes due to Failed to create route payment: Route(payment)[From[kamelet:jms-apache-artemis-source?broker... because of Failed to resolve endpoint: kamelet://jms-apache-artemis-source?brokerURL=tcp://172.28.132.151:61616&destinationName=payments&destinationType=queue due to: org.apache.camel.component.kamelet.KameletNotFoundException: Kamelet with id jms-apache-artemis-source not found in locations: file:/karavan/code,classpath:/kamelets,github:apache:camel-kamelets/kamelets. This exception is ignored.
org.apache.camel.FailedToCreateRouteException: Failed to create route payment: Route(payment)[From[kamelet:jms-apache-artemis-source?broker... because of Failed to resolve endpoint: kamelet://jms-apache-artemis-source?brokerURL=tcp://172.28.132.151:61616&destinationName=payments&destinationType=queue due to: org.apache.camel.component.kamelet.KameletNotFoundException: Kamelet with id jms-apache-artemis-source not found in locations: file:/karavan/code,classpath:/kamelets,github:apache:camel-kamelets/kamelets
        at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:82) ~[camel-core-reifier-4.1.0.jar:4.1.0]
```
 
 
![image](https://github.com/apache/camel-karavan/assets/33138463/be459e60-32ef-4960-9c0d-ec49696dc710)

When I downloaded in the demo pod the Artemis cli and checked the connection, it show the connection brockerURL with queues  (see snapshot below)

![image](https://github.com/apache/camel-karavan/assets/33138463/56e81156-7137-408e-9885-b94afbeaa34c)

Or when I build a demo container, I get below error (see snapshot below) :

![image](https://github.com/apache/camel-karavan/assets/33138463/e842cf8d-cb38-4ee5-9f02-30be0aded5e5)

_______________________________
Please help with this issue.

**Comment by mgubaidullin** [comment]
@davsclaus I can confirm that the following example  
```
- route:
    id: route-f0b7
    nodePrefixId: route-d25
    from:
      id: from-130a
      uri: kamelet:jms-apache-artemis-source
      parameters:
        destinationName: test
        destinationType: queue
        brokerURL: tcp://localhost:61616
      steps:
        - log:
            id: log-e983
            message: ${body}

```
throws exception with camel-jbang
```
Error starting CamelContext (jms-demo) due to exception thrown: Failed to start route jms-apache-artemis-source-1 because of connectionFactory must be specified
org.apache.camel.FailedToStartRouteException: Failed to start route jms-apache-artemis-source-1 because of connectionFactory must be specified
        at org.apache.camel.impl.engine.RouteService.setUp(RouteService.java:131) ~[camel-base-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doWarmUpRoutes(InternalRouteStartupManager.java:307) ~[camel-base-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.safelyStartRouteServices(InternalRouteStartupManager.java:187) ~[camel-base-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRoutes(InternalRouteStartupManager.java:144) ~[camel-base-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartCamel(AbstractCamelContext.java:2778) ~[camel-base-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartContext(AbstractCamelContext.java:2459) [camel-base-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStart(AbstractCamelContext.java:2414) [camel-base-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.support.service.BaseService.start(BaseService.java:113) [camel-api-4.3.0.jar:4.3.0]
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2019) [camel-base-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:208) [camel-core-engine-4.3.0.jar:4.3.0]
        at org.apache.camel.main.KameletMain.doStart(KameletMain.java:323) [camel-kamelet-main-4.3.0.jar:4.3.0]
```

**Comment by davsclaus** [comment]
Its because of javax vs jakartaee - that seems to cause the CF to not be of correct type (jakartaee)

**Comment by davsclaus** [comment]
Okay its the kamelet that uses the wrong dependency - it should use

        <dependency>
            <groupId>org.apache.activemq</groupId>
            <artifactId>artemis-jakarta-client-all</artifactId>
            <version>2.31.2</version>
        </dependency>

**Comment by davsclaus** [comment]
Okay will be fixed in Camel 4.4

**Comment by RafiIsaharov** [comment]
Hi ,

Thank you for letting me know the issue will be fixed in Camel 4.4. I am
looking forward to the update!

Best regards,
Rafi
